core/src/main/java/com/netflix/conductor/core/execution/DeciderService.java

```diff
@@ -331,7 +351,7 @@ private boolean checkForWorkflowCompletion(final Workflow workflow) throws Termi
             }
         }
 
-        String taskReferenceName = task.getReferenceTaskName();
+        String taskReferenceName = task.isLoopOverTask() ? DoWhileTaskMapper.getTaskDefReferenceName(task.getReferenceTaskName()) : task.getReferenceTaskName();
```

Comment:
This looks leaky to me that the specific TaskMapper implementation is explicitly invoked here.